src/sema_manager.cc

```diff
@@ -432,6 +436,8 @@ void *CompletionMain(void *manager_) {
   set_thread_name("comp");
   while (true) {
     std::unique_ptr<SemaManager::CompTask> task = manager->comp_tasks.Dequeue();
+    if (pipeline::quit)
```

Comment:
A full memory barrier is probably not necessary here?

Reply:
`ThreadedQueue::{Dequeue,PushBack}` use a mutex internally, the access of `pipeline::quit` is guaranteed to be synchronized.

Reply:
Yes a seq_cst barrier is an overkill. Consider the following:

```c++
if(pipeline::quit.load(std::memory_order_relaxed)
  ...
````
(or `memory_order_acquire` if it wasn't protected by a mutex).

Reply:
It seems everything is protected by a mutex. I can change every load of `quit` to use `std::memory_order_relaxed`